tests/system/Models/InsertModelTest.php

```diff
@@ -286,4 +288,29 @@ public function testInsertWithSetAndEscape(): void
 
         $this->assertCloseEnough(time(), strtotime($result->created_at));
     }
+
+    public function testInsertWithUpdate(): void
+    {
+        $entity = new User();
+        $this->createModel(UserObjModel::class);
+
+        $entity->name       = 'Mark';
+        $entity->email      = 'mark@example.com';
+        $entity->country    = 'India';
+        $entity->deleted    = 0;
+        $entity->created_at = new Time('now');
+
+        $this->model->insert($entity);
+        $this->assertGreaterThan(0, $this->model->getInsertID());
+        $resultEntity = $this->model->where('name', 'Mark')->first();
+
+        $resultEntity->name = 'Katherine';
+
+        $this->model->insert($resultEntity);
+        $this->assertGreaterThan(0, $this->model->getInsertID());
+
+        $resultEntity2 = $this->model->where('name', 'Katherine')->first();
+        $this->assertGreaterThan(0, $this->model->getInsertID());
+        $this->assertSame('mark@example.com', $resultEntity2->email);
+    }
```

Comment:
I'm sorry, but I don't understand this test's intention.

Test code needs to be very easily to understand.

And the code works without errors, but it seem it is a wrong usage of Model/Entity.
You create an Entity with name Mark.
And you changed the name to Katherine.
I don't imagine that Mark becomes Katherine.

Reply:
This PR is to fix #4247, and it has sample code like:
```php
$bugExample = new BugExample();
$bugExample->setTheAttribute(500);
$this->insert($bugExample);
// The insert query will not include the_attribute
```
Why don't you write test like it?

Reply:
@kenjis, I'm sorry if I have done this incorrectly.

The only way I have seen the bug appear is when doing a write (insert) read back the value and the write (insert) the original row with a change. In this case, only the changed item is saved back. This is what the test checks for. I have run the test with the old code, and tested the old path and it fails.

So the test, while not perhaps obvious is the code that provokes the error and tests that it is fixed.

Now, there might be other ways of testing for it, but I don't know what they are. This is what I was doing when it showed up.

As I understand it, the code must be run with a non-Standard Class object, which means I need to use a Model and Entity. Once again, this might be my lack of understanding, but I don't know how else to do it.

Reply:
> So the test, while not perhaps obvious is the code that provokes the error and tests that it is fixed.

Yes, you got the bug and fixed it. And your test failed before the fix.
It is very good practice that writing a test that detects a bug, and fix the bug.

> As I understand it, the code must be run with a non-Standard Class object, which means I need to use a Model and Entity.

Yes, you're correct. This is a bug related with `Entity::toRawArray()`.

I suggested the test code referring to #4247.